ams/server/src/main/java/com/netease/arctic/server/catalog/InternalIcebergCatalogImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.netease.arctic.server.catalog;
+
+import com.netease.arctic.ams.api.CatalogMeta;
+import com.netease.arctic.catalog.IcebergCatalogWrapper;
+import com.netease.arctic.io.ArcticFileIO;
+import com.netease.arctic.io.ArcticFileIOAdapter;
+import com.netease.arctic.server.iceberg.InternalTableOperations;
+import com.netease.arctic.server.persistence.mapper.TableMetaMapper;
+import com.netease.arctic.server.table.TableMetadata;
+import com.netease.arctic.server.utils.IcebergTableUtils;
+import com.netease.arctic.table.ArcticTable;
+import org.apache.iceberg.BaseTable;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.io.FileIO;
+
+public class InternalIcebergCatalogImpl extends MixedCatalogImpl {
```

Comment:
Why should the `InternalIcebergCatalogImpl` extend `MixedCatalogImpl`?

Reply:
to reuse the logic of  listDatabase,   exist(database)   exist(table)

Reply:
Extract the reuse able method to `InternalCatalog` and make the `InternalIcebergCatalogImpl` to extend `InternalCatalog`